action.yml

```diff
@@ -20,6 +20,17 @@ inputs:
     description: "Cache key to use for retaining the Nix store"
   existing-nix:
     description: "If we should utilize the pre-installed Nix instead of adding our own"
+  substituter:
+    description: "Default substituter to use for uploads"
+  substituter-options:
+    description: "Default substituter options"
+    default: "?secret-key=/tmp/secret-key&write-nar-listing=1&ls-compression=br"
+  substituter-key:
+    description: "Substituter key created by `nix key generate-secret`"
+  aws-access-key-id:
+    description: "AWS access key to upload"
+  aws-secret-access-key:
+    description: "AWS secret key to upload"
```

Comment:
could this be done in a more abstract way? I think it would be nicer if we don't have to have "aws" in this anywhere and specific hardcoded env vars, and instead any could be provided

Reply:
i'm open to ideas.... this approach makes it nicely symmetrical with the rest of the config.... otherwise we can put it into a yaml section allowing for more flexibility and less hard-coding.
```
env: 
  AWS_ACCESS_KEY_ID=${{ secrets.MY_AWS_ID }}
```

Reply:
resolving for now as all the s3-compatible stores will use aws libs